subsystem/src/manager.rs

```diff
@@ -364,16 +364,22 @@ impl Manager {
 
 /// Used to initiate shutdown of manager and subsystems.
 #[derive(Clone)]
-pub struct ShutdownTrigger(mpsc::Sender<()>);
+pub struct ShutdownTrigger(mpsc::WeakSender<()>);
 
 impl ShutdownTrigger {
     /// Initiate shutdown
     pub fn initiate(&self) {
         use mpsc::error::TrySendError as E;
-        self.0.try_send(()).unwrap_or_else(|err| match err {
-            E::Full(_) => log::info!("Shutdown requested but the system is already shutting down"),
-            E::Closed(_) => log::info!("Shutdown requested but the system is already down"),
-        })
+        if let Some(sender) = self.0.upgrade() {
+            sender.try_send(()).unwrap_or_else(|err| match err {
+                E::Full(_) => {
+                    log::info!("Shutdown requested but the system is already shutting down")
+                }
+                E::Closed(_) => log::info!("Shutdown requested but the system is already down"),
+            })
+        } else {
+            log::info!("Shutdown requested but the system is already down")
+        }
```

Comment:
I'm not sure this is better, but it allows not to repeat the message:
```suggestion
match self.0.upgrade().map(|s| s.try_send(())) {
            None | Some(Err(E::Closed(_))) => {
                log::info!("Shutdown requested but the system is already down")
            }
            Some(Err(E::Full(_))) => {
                log::info!("Shutdown requested but the system is already shutting down")
            }
            Some(Ok(())) => {}
        }
```